lightwood/mixers/helpers/default_net.py

```diff
@@ -34,11 +36,21 @@ def __init__(self, dynamic_parameters,
         super(DefaultNet, self).__init__()
 
         if shape is None and pretrained_net is None:
-            shape = [self.input_size, max([self.input_size*2,self.output_size*2,400]), self.output_size]
+            # default shape, inflated, yields 240k params if input has 200 components
+            shape = [self.input_size, max([self.input_size*2, self.output_size*2, 400]), self.output_size]
+
+            # if NN is too big, we do not inflate and aim for max target_params
+            if reduce(lambda x, y: x*y, shape) > target_params:
+                hidden_size = int(target_params//(self.input_size*self.output_size))
+
+                if hidden_size > self.output_size:
+                    shape = [self.input_size, hidden_size, self.output_size]
+                else:
+                    shape = [self.input_size, self.output_size]
```

Comment:
The way this scales down is too aggressive.
Instead have  `while reduce...` and every loop:

`if hidden_size > self.output_size` => `hidden_size=hidden_size*0.99`

else: `[self.input_size, self.output_size]` and break (though if the data is so big we might as well just "quit" and rely on the lightGBM mixer maybe?, to think about for later though)